pkg/logs/input/docker/ad_identifier_kubelet.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build docker,kubelet
+
+package docker
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
+)
+
+// configPath refers to the configuration that can be passed over a docker label or a pod annotation,
+// this feature is commonly named 'ad' or 'autodicovery'.
+const (
+	labelConfigPath            = "com.datadoghq.ad.logs"
+	annotationConfigPathPrefix = "ad.datadoghq.com"
+	annotationConfigPathSuffix = "logs"
+)
+
+// ContainsADIdentifier returns true if the container contains an autodiscovery identifier.
+func ContainsADIdentifier(c *Container) bool {
+	var exists bool
+	_, exists = c.container.Labels[labelConfigPath]
+	if exists {
+		return true
+	}
+	kubeutil, err := kubelet.GetKubeUtil()
+	if err != nil {
+		return false
+	}
+	pod, err := kubeutil.GetPodForEntityID(c.service.GetEntityID())
+	if err != nil {
+		return false
+	}
+	for _, container := range pod.Status.Containers {
```

Comment:
why do we loop over pod's containers, instead of using `c`? worth an explanation